tools/project_manager/external_catalog_search.php

```diff
@@ -129,186 +217,227 @@ function do_search_and_show_hits()
+    $next = $start + $hits_per_page;            // next page will be 61 to 70
+    $prev = max(1, $start - $hits_per_page);    // prev page will be 41 to 50. If we start at <11, prev=1
+    $last = $total_hits - ($hits_per_page - 1); // last page will be 91 to 100.
+    $frags = [
+        link_or_text($url_base, "First", $start > 1, 1),
+        link_or_text($url_base, "Previous", $start > 1, $prev),
+        sprintf(_("Results %d to %d of %d%s"), $start, min($total_hits, $next - 1), $total_hits, $book_frag),
+        link_or_text($url_base, "Next", $next <= $total_hits, min($total_hits, $next)),
+        link_or_text($url_base, "Last", $next <= $total_hits, $last),
+    ];
+    echo "<p class='center-align'>", implode(" | ", $frags), "</p>\n";
+}
 
-    echo "</p>";
-    echo "</form>";
-    yaz_close($id);
+function display_record_table(MARCRecord $marc_record, bool $hide_nontext): void
+{
+    echo "<table class='basic' style='width: 100%;'>";
+    $fields = [];
+    if (!$hide_nontext) {
+        $fields[] = [_("Type of Record"), $marc_record->type_of_record];
+    }
+    $fields = array_merge($fields, [
+        [_("Title"),     $marc_record->title],
+        [_("Author"),    $marc_record->author],
+        [_("Publisher"), $marc_record->publisher],
+        [_("Language"),  $marc_record->language],
+        [_("LCCN"),      $marc_record->lccn],
+        [_("ISBN"),      $marc_record->isbn],
+    ]);
+    foreach ($fields as [$label, $value]) {
+        echo "<tr>";
+        echo   "<th class='left-align top-align' style='width: 20%;'>{$label}:</th>";
+        echo   "<td class='left-align top-align'>{$value}</td>";
+        echo "</tr>\n";
+    }
+    echo "</table>";
 }
 
 // XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 
 function query_format()
 {
-    $attr_set = 0;
-    $fullquery = "";
+    global $title_attrs, $author_attrs, $serial_attrs;
+    // Build a Z39.50 Type-1 query.
+    // See
+    // https://www.loc.gov/z3950/agency/markup/09.html Type-1 and Type-101 Queries
+    //
+    //
+    // A Type-1 query is in prefix polish notation and a (prettified) simple query
+    // for editions of Snowcrash published >= 1992 might look like
+    // ```
+    //  @and
+    //  @and
+    //  @attr 1=6 "Snowcrash"
+    //  @attr 1=1004 "Stephenson, Neal"
+    //  @attr 2=4 @attr 1=31 1992
+    // ```
+    // Expressions like `@attr 1=1004` mean value 1004 from attribute set 1
+    // Set 1 is "Bib-1 Use", and 1004 is the value "Author-name-personal"
+    // https://www.loc.gov/z3950/agency/defns/bib1.html
+    // https://www.loc.gov/z3950/agency/bib1.html
+
+    $fullquery = [];
 
     if ($_REQUEST['title']) {
-        $fullquery = $fullquery.' @attr 1=4 "'.$_REQUEST['title'].'"';
-        $attr_set++;
+        $attr = get_enumerated_param($_REQUEST, 'title_attr', null, array_keys($title_attrs));
+        $fullquery[] = '@attr 1=' . $attr . ' "' . $_REQUEST['title'] . '"';
```

Comment:
tip: using `sprintf()` would probably make this easier to read.

Reply:
Ditto the others in this function.

Reply:
Now that you say it, it probably would! Done.